File: 0647-palindromic-substrings/0647-palindromic-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
const countSubstrings = s => {
    let total = 0;
    
    const count = (s, i, j) => {
        let count = 0;
        while (i >= 0 && j < s.length && s[i] === s[j]) {
            i--;
            j++;
            count++;
        }
        return count;
    }
    
    for (let i = 0; i < s.length; i++) {
        total += count(s, i, i);
        total += count(s, i, i + 1);
    }
    return total;
};

File: 0647-palindromic-substrings/0647-palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        total = 0
        
        def count(i, j):
            count = 0
            while i >= 0 and j < len(s) and s[i] == s[j]:
                i -= 1
                j += 1
                count += 1
            return count
        
        for i in range(len(s)):
            total += count(i, i)
            total += count(i, i + 1)
        return total